lib/LedgerSMB/Report/Listings/TemplateTrans.pm

```diff
@@ -110,18 +110,14 @@ sub name {
 
 =cut
 
-my %jtype = (
-    1 => 'gl',
-    2 => 'ar',
-    3 => 'ap',
-    );
+my @jtype = ( undef, 'gl', 'ar', 'ap' );    # XXX magic number map.  From where is the magic? --rir
```

Comment:
Hmm, Rather than xxx, please use "fixme: " or "todo: " with an explanation of what the issue is and what may need to be done to resolve it.
That way some of the automated tooling around can easily identify things that need to be addressed.
The use of "xxx" would simply just get overlooked forever.

As to the source of the magic, it's possible that it comes from very old legacy constructs that have resulted in these constant numbers being stored in the DB, @ehuelsmann may be able to cast more light on that, and the best way to handle it as well